File: Backend-EmploiConnect/src/main/java/com/emploiconnect/dto/response/ApplicationResponseDto.java
package com.emploiconnect.dto.response;

import com.emploiconnect.enums.ApplicationStatus;
import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ApplicationResponseDto {
    private Long id;

    private ApplicationStatus status;

    @NotNull(message = "Title cannot be null")
    private String title;
    private String cv;
    private String profile;


    private OfferResponseDto offer;
    private AuthenticationResponse user;
}

File: FrontEnd-EmploiConnect/src/app/dtos/responses/ApplicationResponseDTO.ts
import { ApplicationStatus } from "src/app/models/enums/ApplicationStatus";
import { AuthenticationResponseDTO } from "./AuthenticationResponseDTO";
import { OfferResponseDTO } from "./OfferResponseDTO";

export interface ApplicationResponseDTO{
    id:number;
    status: ApplicationStatus;
    title:String;
    profile:String;
    offer: OfferResponseDTO;
    user: AuthenticationResponseDTO;
    cv:String;
}

